src/server/server.ts

```diff
@@ -598,7 +598,7 @@ namespace ts.server {
             if (option && value) {
                 switch (option) {
                     case "-file":
-                        logEnv.file = stripQuotes(value);
+                        logEnv.file = getLogPath(args, i);

```

Comment:
If `file` comes first and has multiple parts, you don't skip over those parts (i.e. you pass the start of the file parts, but you still only increment i by 2 on each for loop, whereas getLogPath may have consumed multiple elements to form the path.

Reply:
Changed the behavior so that the extra parts of the file path are consumed rather than reiterated so that we don't get off-cycle when iterating by 2.